src/core/ApolloClient.ts

```diff
@@ -511,10 +512,38 @@ export class ApolloClient implements DataProxy {
   >(
     options: MutationOptions<TData, TVariables, TCache>
   ): Promise<MutateResult<MaybeMasked<TData>>> {
-    if (this.defaultOptions.mutate) {
-      options = mergeOptions(this.defaultOptions.mutate, options);
+    const optionsWithDefaults = mergeOptions(
+      compact(
+        {
+          fetchPolicy: "network-only" as MutationFetchPolicy,
+          errorPolicy: "none" as ErrorPolicy,
+        },
+        this.defaultOptions.mutate
+      ),
+      options
+    ) as MutationOptions<TData, TVariables, TCache> & {
+      fetchPolicy: MutationFetchPolicy;
+      errorPolicy: ErrorPolicy;
+    };
+
+    if (__DEV__) {
+      invariant(
+        optionsWithDefaults.mutation,
+        "mutation option is required. You must specify your GraphQL document in the mutation option."
+      );
+
+      invariant(
+        optionsWithDefaults.fetchPolicy === "network-only" ||
+          optionsWithDefaults.fetchPolicy === "no-cache",
+        "Mutations support only 'network-only' or 'no-cache' fetchPolicy strings. The default `network-only` behavior automatically writes mutation results to the cache. Passing `no-cache` skips the cache write."
```

Comment:
```suggestion
        "Mutations only support 'network-only' or 'no-cache' fetch policies. The default 'network-only' behavior automatically writes mutation results to the cache. Passing 'no-cache' skips the cache write."
```

Consistency with the first part of the message and some small tweaking (I know you didn't write the original, but doesn't hurt to tweak now!) 🙂